python/ray/tests/test_chaos.py

```diff
@@ -123,6 +127,92 @@ def get(self):
     # assert_no_system_failure(p, 10)
 
 
+@pytest.fixture
+def set_kill_interval(request):
+    lineage_reconstruction_enabled, kill_interval = request.param
+    os.environ["RAY_lineage_pinning_enabled"] = ("1" if
+                                                 lineage_reconstruction_enabled
+                                                 else "0")
+    request.param = {
+        "kill_interval": kill_interval,
+        "head_resources": {
+            "CPU": 1
+        },
+        "worker_node_types": {
+            "cpu_node": {
+                "resources": {
+                    "CPU": 2,
+                },
+                "node_config": {},
+                "min_workers": 0,
+                "max_workers": 4,
+            },
+        },
+        # We just want the autoscaler to restart nodes, not take them down.
+        "idle_timeout_minutes": 10,
+    }
+    cluster = _ray_start_chaos_cluster(request)
+    while True:
+        try:
+            yield (lineage_reconstruction_enabled, kill_interval,
+                   next(cluster))
+        except StopIteration:
+            break
+    del os.environ["RAY_lineage_pinning_enabled"]
+
+
+@pytest.mark.parametrize(
+    "set_kill_interval", [(True, None), (True, 30), (False, None),
+                          (False, 30)],
+    indirect=True)
+def test_nonstreaming_shuffle(set_kill_interval):
+    lineage_reconstruction_enabled, kill_interval, _ = set_kill_interval
+    try:
+        # Create our own tracker so that it gets scheduled onto the head node.
+        tracker = shuffle._StatusTracker.remote()
+        ray.get(tracker.get_progress.remote())
+        assert len(ray.nodes()) == 1, (
+            "Tracker actor may have been scheduled to remote node "
+            "and may get killed during the test")
+
+        shuffle.run(
+            ray_address="auto",
+            no_streaming=True,
+            num_partitions=200,
+            partition_size=10e6,
+            tracker=tracker)
+    except (RayTaskError, ObjectLostError):
+        assert kill_interval is not None
+        assert not lineage_reconstruction_enabled
+
+
+@pytest.mark.parametrize(
+    "set_kill_interval", [(True, None), (True, 30), (False, None),
+                          (False, 30)],
+    indirect=True)
+def test_streaming_shuffle(set_kill_interval):
```

Comment:
Is this supposed to work? (Isn't the owner distributed workload, which is not working with lineage reconstruction? )

Reply:
This is currently checking that the application gets the right error (i.e., system doesn't crash).

Reply:
Ah this makes sense